src/sql/workbench/contrib/notebook/browser/notebookViews/insertCellsModal.ts

```diff
@@ -153,41 +146,24 @@ export class InsertCellsModal extends Modal {
 		const activeView = this._context.getActiveView();
 		const cellsAvailableToInsert = activeView.hiddenCells;
 
-		cellsAvailableToInsert.forEach(async (cell) => {
-			const optionWidget = this.createCheckBoxHelper(
-				container,
-				'<div class="loading-spinner-container"><div class="loading-spinner codicon in-progress"></div></div>',
-				false,
-				() => this.onOptionChecked(cell.cellGuid)
-			);
-
-			const img = await this.generateScreenshot(cell);
-			const wrapper = DOM.$<HTMLDivElement>('div.thumnail-wrapper');
-			const thumbnail = DOM.$<HTMLImageElement>('img.thumbnail');
-
-			thumbnail.src = img;
-			thumbnail.style.maxWidth = '100%';
-			DOM.append(wrapper, thumbnail);
-			optionWidget.label = wrapper.outerHTML;
-
-			this._optionsMap[cell.cellGuid] = optionWidget;
-		});
-	}
+		const thumbnails = await Promise.all(
```

Comment:
Consider adding logic to handle errors separately so that one thumbnail failing to be generated doesn't break the whole dialog.

Reply:
Added some really simple error handling that results in a blank card in case of an error. I'll work on displaying an error message in the thumbnail.